Android.mk

```diff
@@ -0,0 +1,308 @@
+    src/core/common/message.cpp                \
+    src/core/common/settings.cpp               \
+    src/core/common/string.cpp                 \
+    src/core/common/tasklet.cpp                \
+    src/core/common/timer.cpp                  \
+    src/core/common/tlvs.cpp                   \
+    src/core/common/trickle_timer.cpp          \
+    src/core/crypto/aes_ccm.cpp                \
+    src/core/crypto/aes_ecb.cpp                \
+    src/core/crypto/hmac_sha256.cpp            \
+    src/core/crypto/mbedtls.cpp                \
+    src/core/crypto/pbkdf2_cmac.cpp            \
+    src/core/crypto/sha256.cpp                 \
+    src/core/mac/mac.cpp                       \
+    src/core/mac/mac_filter.cpp                \
+    src/core/mac/mac_frame.cpp                 \
+    src/core/meshcop/announce_begin_client.cpp \
+    src/core/meshcop/commissioner.cpp          \
+    src/core/meshcop/dataset.cpp               \
+    src/core/meshcop/dataset_local.cpp         \
+    src/core/meshcop/dataset_manager.cpp       \
+    src/core/meshcop/dataset_manager_ftd.cpp   \
+    src/core/meshcop/dtls.cpp                  \
+    src/core/meshcop/energy_scan_client.cpp    \
+    src/core/meshcop/joiner.cpp                \
+    src/core/meshcop/joiner_router.cpp         \
+    src/core/meshcop/leader.cpp                \
+    src/core/meshcop/meshcop.cpp               \
+    src/core/meshcop/meshcop_tlvs.cpp          \
+    src/core/meshcop/panid_query_client.cpp    \
+    src/core/meshcop/timestamp.cpp             \
+    src/core/net/dhcp6_client.cpp              \
+    src/core/net/dhcp6_server.cpp              \
+    src/core/net/dns_client.cpp                \
+    src/core/net/icmp6.cpp                     \
+    src/core/net/ip6.cpp                       \
+    src/core/net/ip6_address.cpp               \
+    src/core/net/ip6_filter.cpp                \
+    src/core/net/ip6_headers.cpp               \
+    src/core/net/ip6_mpl.cpp                   \
+    src/core/net/ip6_routes.cpp                \
+    src/core/net/netif.cpp                     \
+    src/core/net/udp6.cpp                      \
+    src/core/thread/address_resolver.cpp       \
+    src/core/thread/announce_begin_server.cpp  \
+    src/core/thread/announce_sender.cpp        \
+    src/core/thread/child_table.cpp            \
+    src/core/thread/data_poll_manager.cpp      \
+    src/core/thread/energy_scan_server.cpp     \
+    src/core/thread/key_manager.cpp            \
+    src/core/thread/link_quality.cpp           \
+    src/core/thread/lowpan.cpp                 \
+    src/core/thread/mesh_forwarder.cpp         \
+    src/core/thread/mesh_forwarder_ftd.cpp     \
+    src/core/thread/mesh_forwarder_mtd.cpp     \
+    src/core/thread/mle.cpp                    \
+    src/core/thread/mle_router.cpp             \
+    src/core/thread/network_data.cpp           \
+    src/core/thread/network_data_leader.cpp    \
+    src/core/thread/network_data_leader_ftd.cpp \
```

Comment:
nit: can we align the backslash - can make it further out so that new files are less likely to cause adjusting all the lines again.